package/src/components/AutoCompleteInput/AutoCompleteSuggestionItem.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { Avatar } from '../Avatar/Avatar';
+import { AtMentions } from '../../icons/AtMentions';
+import type { DefaultUserType } from '../../types/types';
+import type { SuggestionUser } from '../../contexts/suggestionsContext/SuggestionsContext';
+import { AutoCompleteSuggestionCommandIcon } from './AutoCompleteSuggestionCommandIcon';
+
+export type CommandItemType = {
+  args?: string;
+  name?: string;
+};
+
+export type EmojiItemType = {
+  name: string;
+  unicode: string;
+};
+
+export type AutoCompleteSuggestionItemPropsWithContext<
+  Us extends DefaultUserType = DefaultUserType,
+> = {
+  itemProps: CommandItemType | EmojiItemType | SuggestionUser<Us>;
+  type?: string;
```

Comment:
After looking at the example in the documentation, I definitely we should either make the trigger type typed to either an enum, or `type TriggerType = "command" | "emoji" | "mention"`, for a better developer experience when users write their cusom implementations. Their editor/IDE will tell them the possible trigger types, and give them an error if they mistype it or check for one that doesn't exist, for example.